packages/bpk-component-calendar/src/BpkCalendarDate.tsx

```diff
@@ -185,6 +186,7 @@ class BpkCalendarDate extends PureComponent<Props> {
     }
 
     delete buttonProps.preventKeyboardFocus;
+    delete buttonProps.formattedDate;
```

Comment:
The returned component includes `{...buttonProps}`, so the html includes `formattedDate="2024-12-19"`. This isn't an issue but:
- wasn't intentional
- invalidates all the existing snapshot tests

Reply:
In my case it is used in the Flights Search Controls GC - https://github.skyscannertools.net/skyscanner/global-components/pull/1607/files

Reply:
I still don't understand, the prop is passed into the component, it's not referenced anywhere nor is `this.props` passed into anything then its deleted from `buttonProps` before it's used. How does including this prop in this component do anything?

Reply:
Apologies. The Global Component uses a custom Date component:
[composeCalendar(null, BpkCalendarGridHeader, BpkScrollableCalendarGridList, Date)](https://github.skyscannertools.net/skyscanner/global-components/blob/507ead9289667f708f88c2052e195662cfc008b0/src/flights-search-controls/components/DatePicker/MobileDatePicker/Calendar.tsx#L39).

We pass the formattedDate to the Date component, in the case of the GC this value is valuable, we use it to [prevent duplicate generation of iso formatted dates](https://github.skyscannertools.net/skyscanner/global-components/pull/1607/files#r1041067) (these dates are used to colour code dates with prices).

In the case of the default Backpack Date component that formattedDate value is not useful, I delete it from the props to avoid it appearing in the HTML and snapshots.